test/test.js

```diff
@@ -7,114 +9,167 @@ var UploadStream = require('../lib/UploadStream');
+    t.equal(processedRecords.length, 2, 'Loads data after filter.');
+  });
+
+  retriever({quiet: true, profile: 'default', directory: 'test/output', file: 'test/data/parcelsjson.json'}, function(errs, processedRecords){
+    t.equal(errs.length, 0, 'No error on converted parcels.')
+    t.equal(processedRecords.length, 1, 'Loads data from parcels');
+  });
+
+  spawn('./retriever.js', ['-b', 'wyatt-test', '-p', 'default', '-d', '.', '-f', maine])
     .on('exit', function(code){
       t.equal(code, 0, 'Loads via cli');
     });
 
-  spawn('test/no-cb.js', ['-b', 'wyatt-test', '-p', 'default', '-d', '.', '-f', maine]) 
+  spawn('./test/no-cb.js', ['-b', 'wyatt-test', '-p', 'default', '-d', '.', '-f', maine])
     .on('exit', function(code){
       t.equal(code, 0, 'Works without a callback.');
     });
-  
-  retriever({profile:'default', directory:'test/output', file:maine}, function(err, count){
-    t.notOk(err, 'No error on good file.');
-    t.equal(count, 1, 'Loads data from test data locally.')
-  }); 
-  
-  retriever({bucket:'wyatt-test', profile:'default', directory:'.', file: maine, match: 'maine'}, function(err, count){
-    t.notOk(err, 'No error with match.');
-    t.equal(count, 1, 'Loads matched data.');
-  });
 
-  retriever({bucket:'wyatt-test', profile:'default', directory:'.', file: maine, match: 'nomatch'}, function(err, count){
-    t.notOk(err, 'No error with no match.');
-    t.equal(count, 0, 'Loads nothing when no data matched.');
-  });
+});
 
-  retriever({profile:'default', directory:'test/output', file: 'test/data/mainejson.json'}, function(err, count){
-    t.notOk(err, 'No error on good json file.');
-    t.equal(count, 1, 'Loads data from json file.')
-  });
+test('Ensure output', function(t){
+  t.plan(8);
 
-  retriever({profile:'default', directory:'test/output', file: 'test/data/mainecsv.json'}, function(err, count){
-    t.notOk(err, 'No error on csv.');
-    t.equal(count, 1, 'Loads data from csv.')
-  });
+  var outfiles = [
+    {file: 'test/output/arkansas.csv.gz', osxhash: '2e50e44d42b2c1ab7aa22d3f1c704ee127298f409deb0a2fddbff49dfd5aebbe', ubuntuhash: '8b76792518342b0c557d5c948b8a282625936086ab7ddeaa394662dab120b1e6'},
+    {file: 'test/output/maine.csv.gz', osxhash: 'aefe30bd7b08afb745a62aa87d0bb9f4d98734d958e25891e0ac4ef31397edfb', ubuntuhash: 'aefe30bd7b08afb745a62aa87d0bb9f4d98734d958e25891e0ac4ef31397edfb'},
+    {file: 'test/output/north_carolina.csv.gz', osxhash: '94795e123dc028d643db2dba749816d17ad4b222b0d6686c44b4e406ad98bf56', ubuntuhash: '318ea9987f7f3e3208026ca589ccbd1076d6aa219c286068556d8f9fcbf0e324'},
+    {file: 'test/output/sacramento.csv.gz', osxhash: '486c0dba103103fbaa87e2a74a5457a724f0ed3f0af8b6c0bdef6254752a39c4', ubuntuhash: '5106df46f78f9a9af787d4c523cbfacf05dba746e1f7a9c62723dc8caa04acf2'}
+  ];
+
+  outfiles.forEach(function(obj){
+    var stream = fs.createReadStream(obj.file);
+    var hash = os.platform() === 'darwin' ? 'osxhash' : 'ubuntuhash';
```

Comment:
I figured this out recently and have it fixed in my current branch. The results of the gzip operation were returning different files and as such were getting different hashes. I initially thought this was a system-dependent gzip thing, but eventually figured out that it had to do with the files not being fully written when they were used in this test (the previous test exited but the async file i/o was still happening). The fix is to ensure that the files are in their final form before running this test (or check in static files rather than working with ones generated earlier by the test).

So worry not, this fix will show up in a different PR.